src/components/CodeShower.vue

```diff
@@ -65,9 +65,9 @@ let text =
 @inputs EGP:wirelink ${this.needUserInteraction() ? "User:entity" : ""}
 @persist ActiveScreen:number
 
-if(first() | duped() | dupefinished()) {
```

Comment:
`duped()` is needed if the E2 chip is spawned with the [duplicator](https://wiki.facepunch.com/gmod/duplicator) as `dupefinished()` is only guaranteed to be called if spawned by the [Advanced Duplicator 2](https://steamcommunity.com/sharedfiles/filedetails/?id=773402917) (or 1). (We also can't have only `duped()` as we want to call this once everything is [spawned](https://github.com/wiremod/wire/wiki/E2%3A-Triggers#dupe))

Reply:
> `duped()` is needed if the E2 chip is spawned with the [duplicator](https://wiki.facepunch.com/gmod/duplicator) as `dupefinished()` is only guaranteed to be called if spawned by the [Advanced Duplicator 2](https://steamcommunity.com/sharedfiles/filedetails/?id=773402917) (or 1). (We also can't have only `duped()` as we want to call this once everything is [spawned](https://github.com/wiremod/wire/wiki/E2%3A-Triggers#dupe))

Yeah but if the E2 gets pasted before others components (like the EGP), and we try to access them this will cause a crash. (common problem on nxserv btw).

Reply:
> Yeah but if the E2 gets pasted before others components (like the EGP), and we try to access them this will cause a crash. (common problem on nxserv btw).

Very true and very unfortunate but if we favor [Advanced Duplicator 2](https://steamcommunity.com/sharedfiles/filedetails/?id=773402917) users by removing `duped()` then [duplicator](https://wiki.facepunch.com/gmod/duplicator) users would be penalized (and vice versa).

Reply:
I just tested it in game and dupefinished() does return true when using the vanilla duplicator (as well as the advanced duplicator).

Try it
```lua
if(dupefinished())
{
    print("dupefinished()")
}

if(duped())
{
    print("duped()")   
}
```

Reply:
> I just tested it in game and dupefinished() does return true when using the vanilla duplicator (as well as the advanced duplicator).

That's a good news, we can say _adios_ to `duped()`